File: 82. Remove Duplicates from Sorted List II/82. Remove Duplicates from Sorted List II.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func deleteDuplicates(head *ListNode) *ListNode {
    if head == nil || head.Next == nil {
        return head
    }
    
    dummy := &ListNode{Val: 0, Next: head}
    prev := dummy
    current := head
    
    for current != nil && current.Next != nil {
        if current.Val == current.Next.Val {
            duplicateValue := current.Val
            for current != nil && current.Val == duplicateValue {
                current = current.Next
            }
            prev.Next = current
        } else {
            prev = current
            current = current.Next
        }
    }
    
    return dummy.Next
}

File: 82. Remove Duplicates from Sorted List II/82. Remove Duplicates from Sorted List II.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode DeleteDuplicates(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        
        ListNode dummy = new ListNode(0, head);
        ListNode prev = dummy;
        ListNode current = head;
        
        while (current != null && current.next != null) {
            if (current.val == current.next.val) {
                int duplicateValue = current.val;
                while (current != null && current.val == duplicateValue) {
                    current = current.next;
                }
                prev.next = current;
            } else {
                prev = current;
                current = current.next;
            }
        }
        
        return dummy.next;
    }
}